idb/postgres/postgres.go

```diff
@@ -358,43 +371,65 @@ func (db *IndexerDb) getMetastate(key string) (jsonStrValue string, err error) {
 
 const setMetastateUpsert = `INSERT INTO metastate (k, v) VALUES ($1, $2) ON CONFLICT (k) DO UPDATE SET v = EXCLUDED.v`
 
-func (db *IndexerDb) setMetastate(key, jsonStrValue string) (err error) {
-	_, err = db.db.Exec(setMetastateUpsert, key, jsonStrValue)
-	return
-}
-
-// If `tx` is null, make a standalone query.
-func (db *IndexerDb) getMaxRoundAccounted(tx *sql.Tx) (round uint64, err error) {
-	query := `select v->>'account_round' from metastate where k = 'state'`
-
-	var row *sql.Row
+// If `tx` is nil, use a normal query.
+func (db *IndexerDb) setMetastate(tx *sql.Tx, key, jsonStrValue string) (err error) {
 	if tx == nil {
-		row = db.db.QueryRow(query)
+		_, err = db.db.Exec(setMetastateUpsert, key, jsonStrValue)
 	} else {
-		row = tx.QueryRow(query)
+		_, err = tx.Exec(setMetastateUpsert, key, jsonStrValue)
 	}
+	return
+}
 
-	var nullableRound sql.NullInt64
-	err = row.Scan(&nullableRound)
+// Returns idb.ErrorNotInitialized if uninitialized.
+// If `tx` is nil, use a normal query.
+func (db *IndexerDb) getImportState(tx *sql.Tx) (importState, error) {
```

Comment:
nit: this is pretty much the best-case scenario for when to name variables `(state importState, err error)`. It would let you avoid all of the `importState{}` and  `var state importState`.

Reply:
I think it's nice to return an empty `importState{}` in case of an error, so the caller will behave weirdly if it doesn't check for errors, and tests can detect that.

Reply:
I'm not sure what you mean, the named variable would be initialized to `importState{}`

Reply:
Actually, I'm wrong about this case. But look at the previous implementation.
```
// GetImportState is part of idb.IndexerDB
func (db *IndexerDb) GetImportState() (state idb.ImportState, err error) {
	var importStateJSON string
	importStateJSON, err = db.getMetastate(stateMetastateKey)
	if err == sql.ErrNoRows || importStateJSON == "" {
		// no previous state, ok
		err = idb.ErrorNotInitialized
		return
	} else if err != nil {
		err = fmt.Errorf("unable to get import state: %v", err)
		return
	}

	err = encoding.DecodeJSON([]byte(importStateJSON), &state)
	if err != nil {
		err = fmt.Errorf("unable to parse import state: %v", err)
	}
	return
}
```

If `DecodeJSON` fails, it might have already written something to `state`. Then we would return an error with an initialized `state`.